> Thanks for this. Have you verified that all of the loaders updated here are incompatible with `list[str / Path]`?

I double check all the unstructured specific filetype loaders, unlike the base `UnstructuredFileLoader` all 17 of them are incompatible with `list[str / Path]`. If we want these loaders  to be incompatible with `list[str / Path]`, we need to modify the `_get_elements` like `UnstructuredFileLoader`.

https://github.com/langchain-ai/langchain/blob/bbc3e3b2cfc20381b212f8fcb463cab56946ab0c/libs/community/langchain_community/document_loaders/unstructured.py#L215-L228

And `UnstructuredCHMLoader` is also incompatible with `Path` type, I forgot to add init for it. I will add a commit later.

```
Traceback (most recent call last):
  File "/home/marsman1996/afgen/test/./hello.py", line 21, in <module>
    UnstructuredCHMLoader(Path("./STEM_2015_12_08.chm")).load()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/home/marsman1996/afgen/test/.venv/lib/python3.13/site-packages/langchain_core/document_loaders/base.py", line 31, in load
    return list(self.lazy_load())
  File "/home/marsman1996/afgen/test/.venv/lib/python3.13/site-packages/langchain_community/document_loaders/unstructured.py", line 107, in lazy_load
    elements = self._get_elements()
  File "/home/marsman1996/afgen/test/.venv/lib/python3.13/site-packages/langchain_community/document_loaders/chm.py", line 30, in _get_elements
    with CHMParser(self.file_path) as f:  # type: ignore[arg-type]
         ~~~~~~~~~^^^^^^^^^^^^^^^^
  File "/home/marsman1996/afgen/test/.venv/lib/python3.13/site-packages/langchain_community/document_loaders/chm.py", line 48, in __init__
    self.file.LoadCHM(path)
    ~~~~~~~~~~~~~~~~~^^^^^^
  File "/home/marsman1996/afgen/test/.venv/lib/python3.13/site-packages/chm/chm.py", line 216, in LoadCHM
    self.file = chmlib.chm_open(archiveName)
                ~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "/home/marsman1996/afgen/test/.venv/lib/python3.13/site-packages/chm/chmlib.py", line 20, in chm_open
    return _chmlib.chm_open(filename)
           ~~~~~~~~~~~~~~~~^^^^^^^^^^
TypeError: a bytes-like object is required, not 'PosixPath'
```

--- a/libs/community/langchain_community/document_loaders/chm.py
+++ b/libs/community/langchain_community/document_loaders/chm.py
@@ -1,4 +1,5 @@
-from typing import TYPE_CHECKING, Dict, List, Union
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Dict, List, Union
 
 from langchain_community.document_loaders.unstructured import UnstructuredFileLoader
 
@@ -23,6 +24,23 @@ class UnstructuredCHMLoader(UnstructuredFileLoader):
     https://github.com/dottedmag/pychm
     http://www.jedrea.com/chmlib/
     """
+
+    def __init__(
+        self,
+        file_path: Union[str, Path],
+        mode: str = "single",
+        **unstructured_kwargs: Any,
+    ):
+        """
+
+        Args:
+            file_path: The path to the CHM file to load.
+            mode: The mode to use when loading the file. Can be one of "single",
+                "multi", or "all". Default is "single".
+            **unstructured_kwargs: Any kwargs to pass to the unstructured.
+        """
+        file_path = str(file_path)
+        super().__init__(file_path=file_path, mode=mode, **unstructured_kwargs)
 
     def _get_elements(self) -> List:
         from unstructured.partition.html import partition_html

